supybot_fedora/plugin.py

```diff
@@ -168,17 +170,33 @@ def __init__(self, irc):
 
         # caches, automatically downloaded on __init__, manually refreshed on
         # .refresh
-        self.userlist = None
+        self.users = None
+        self.faslist = None
+        self.nickmap = None
 
         # To get the information, we need a username and password to FAS.
         # DO NOT COMMIT YOUR USERNAME AND PASSWORD TO THE PUBLIC REPOSITORY!
         self.fasurl = self.registryValue("fas.url")
         self.username = self.registryValue("fas.username")
         self.password = self.registryValue("fas.password")
 
-        self.fasclient = AccountSystem(
-            self.fasurl, username=self.username, password=self.password
-        )
+        # Use FASJSON
+        if self.registryValue("use_fasjson"):
+            try:
+                self.fasjsonclient = fasjson_client.Client(
+                    url=self.registryValue("fasjson.url"),
+                    principal="admin@EXAMPLE.TEST",
```

Comment:
The principal should not be specified, so it'll pick up the principal from the kerberos ticket.

Reply:
Removed. I think i added this when trying to get kerberos working, but forgot to remove it. Sorry!